### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Phutoast**
Adding the Simple Multi-Agent Environment, including partial observable and fully observable + documentation.

### CODE REVIEW

**Comment by Phutoast** [comment]
This will be covering all #1 features

File: README.md [hunk 1]
```diff
@@ -1,3 +1,42 @@
 # Capture The Flag Gridworld
 
-This repository contains the implementation of multi-agent capture the flag in 2D, gridworld.
+This repository contains the implementation of multi-agent capture the flag in 2D, gridworld. Base on `pycolab`
```
Comment (n00bAI) [inline README.md]: *based

File: gridworld/partialObsEnv.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        left = pos_x-self.window_size
+        up = pos_y-self.window_size
+        right = pos_x+self.window_size+1
+        down = pos_y+self.window_size+1
+
+        image_y, image_x, _ = image.shape
+        right_offset = image_x - right
+        down_offset = image_y - down
+
+        cropped_image = image[max(up, 0):min(image_y, down), max(left, 0):min(image_x, right), :]
```
Comment (n00bAI) [inline gridworld/partialObsEnv.py]: very nicely done!

File: gridworld/dummy/dummyMap.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+            '#     #     #',
+            '#### ###### #',
+            '#     #     #',
+            '# $   #     #',
+            '#   Q       #',
+            '#     #     #',
+            '#P$   #     #',
+            '#############']
+
+COLOR_MAP = {'#': (0, 0, 0), ' ': (1, 1, 1), 'P': (1, 0, 0), 'Q':(0, 0, 1), '$': (0, 1, 0)}
```
Comment (n00bAI) [inline gridworld/dummy/dummyMap.py]: be nice to have the option to randomise this colour map at a later date, to make it more challenging/ stop overfitting.

File: gridworld/dummy/dummyPlayer.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from gridworld.player import Player
+
+class PlayerSprite1(Player):
+    def __init__(self, corner, position, character):
+        self.playerNum = 0
+        super().__init__(corner, position, character)
+
+    def behavior(self, actions, board, the_plot):
+        if ord('$') == board[self.position]:
```
Comment (n00bAI) [inline gridworld/dummy/dummyPlayer.py]: what does ord('$') do here?

File: gridworld/dummy/dummyMap.py (hunk 3 above)
Comment (Phutoast) [reply gridworld/dummy/dummyMap.py]: Let me add it as an feature in issue page

File: gridworld/dummy/dummyPlayer.py (hunk 4 above)
Comment (Phutoast) [reply gridworld/dummy/dummyPlayer.py]: So basically it turns the character `'$'` into the ASCII number, which is represented in `board[self.position]`.

File: gridworld/dummy/dummyMap.py (hunk 3 above)
Comment (Phutoast) [reply gridworld/dummy/dummyMap.py]: Added in #11

**Pull request merged**